src/compiler/checker.ts

```diff
@@ -13881,6 +13881,10 @@ namespace ts {
                         }
                     }
                 }
+                else if (compilerOptions.noImplicitReturns && !isTypeOfKind(returnType, TypeFlags.Void)) {

```

Comment:
With your suggestion, this code is now accepted.  Is that expected?  (I don't have a good intuition about how "any" ought to behave, so I just wanted to double check.)
```    
    function isMissingReturnExpression2(): any {
      return;
    }
```